### ISSUE TITLE
NestJS serve does not work out of the box

### ISSUE THREAD

**Issue by anjildhamala** [nrwl/nx#11740]
## Current Behavior
Nest app fails to serve

## Expected Behavior
I expected the server to run and accessible at localhost:3333

## Steps to Reproduce
Run `yarn add -D @nrwl/nest` on an existing nrwl@14 project.
Add a nest app  `nx g @nrwl/nest:app isdc_service_mock`
Run the app `nx serve isdc-service-mock`

### Failure Logs
```
> nx run isdc-service-mock:serve --verbose=true


 >  NX   Cannot read properties of undefined (reading 'nodes')


TypeError: Cannot read properties of undefined (reading 'nodes')
    at calculateProjectDependencies (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/@nrwl/js/node_modules/@nrwl/workspace/src/utilities/buildable-libs-utils.js:17:30)
    at calculateResolveMappings (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/@nrwl/js/src/executors/node/node.impl.js:61:86)
    at nodeExecutor_1 (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/@nrwl/js/src/executors/node/node.impl.js:37:26)
    at nodeExecutor_1.next (<anonymous>)
    at resume (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/tslib/tslib.js:230:48)
    at /Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/tslib/tslib.js:229:125
    at new Promise (<anonymous>)
    at Object.i.<computed> [as next] (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/tslib/tslib.js:229:67)
    at Object.i.<computed> [as next] (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/tslib/tslib.js:240:99)
    at nodeExecutor_1 (/Users/anjil.dhamala/Documents/work/cobra-platform/node_modules/@nrwl/node/src/executors/node/node.impl.js:9:46)

 —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target "isdc-service-mock:serve" failed

   Failed tasks:
   
   - isdc-service-mock:serve
   
   Hint: run the command with --verbose for more details.

   See Nx Cloud run details at https://nx.app/runs/bWt3yghoVKe
```

### Environment

   Node : 16.15.1
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx : 14.5.4
   @nrwl/angular : 14.5.4
   @nrwl/cypress : 14.5.4
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.5.4
   @nrwl/eslint-plugin-nx : 14.5.4
   @nrwl/express : Not Found
   @nrwl/jest : 14.5.4
   @nrwl/js : 14.5.10
   @nrwl/linter : 14.5.4
   @nrwl/nest : 14.5.10
   @nrwl/next : Not Found
   @nrwl/node : 14.5.10
   @nrwl/nx-cloud : 14.3.0
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.5.4
   @nrwl/web : 14.5.4
   @nrwl/workspace : 14.5.4
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
         @ngrx/component: 14.0.2
         @ngrx/component-store: 14.0.2
         @ngrx/effects: 14.0.2
         @ngrx/entity: 14.0.2
         @ngrx/router-store: 14.0.2
         @ngrx/store: 14.0.2
         rxjs: 6.6.7
         @ngrx/schematics: 14.0.2
         @ngrx/store-devtools: 14.0.2

**Comment by AgentEnder** [comment]
Hey @anjildhamala , some of your @nrwl/* packages don't line up with the version of `nx` you have installed. Aside from `@nrwl/nx-cloud`, all packages under the nrwl scope should generally be set up with the same version. 

Can you correct this and check that the issue still reproduces? An easy way to fix this would be to run `nx migrate 14.5.10`. Alternatively, updating to latest with `nx migrate latest` would also work.

**Comment by anjildhamala** [comment]
> Hey @anjildhamala , some of your @nrwl/* packages don't line up with the version of `nx` you have installed. Aside from `@nrwl/nx-cloud`, all packages under the nrwl scope should generally be set up with the same version.
> 
> Can you correct this and check that the issue still reproduces? An easy way to fix this would be to run `nx migrate 14.5.10`. Alternatively, updating to latest with `nx migrate latest` would also work.

That was exactly it! Thanks for pointing that you. I migrated to the latest minor version of 14 and it works now. Thanks again. Closing the issue...
